### PR TITLE
feat: allow to store derived info and file is optional

### PR DESCRIPTION
**Pull request by maasencioh**
1. If the file exists:
1.1. Use the received filename or search it in the DB
1.2. Saves the file in the Driver space
1.3. Creates the file in a single DB for any measurement kind
2. Create the measurement:
2.1. If the collection name matches with one already known uses that Model
2.2 If the collection is unknown it is saved in a general measurements collection with that collection name as a field

### CODE REVIEW

File: app/Models/Measurement/Transfer.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  };
+  subthresholdSlope: {
+    medianSlope: number;
+    toIndex: number;
+    fromIndex: number;
+  };
+}
+
+export class TransferMeasurement extends BaseMeasurement {
+  public static collectionName = 'measurement.transfer';
```
Comment (targos) [inline app/Models/Measurement/Transfer.ts]: should we use plural? `measurement.transfers` or `measurements.transfer` ?

File: addons/file-sync/routes.ts [hunk 2]
```diff
@@ -4,23 +4,25 @@ import DataDrive from '@ioc:Zakodium/DataDrive';
...
+  response.header(
+    'Content-Disposition',
+    `attachment; filename="${file.filename}"`,
+  );
+  return response.stream(content);
+}
+
+Route.group(() => {
+  Route.get('/file-content', fetchFile);
+  Route.post('/file-content', fetchFile);
```
Comment (targos) [inline addons/file-sync/routes.ts]: I don't understand what's the goal of the post vs get?

Comment (maasencioh) [reply addons/file-sync/routes.ts]: I'm sending in the post request for the token, so I don't want it to be public, but the get route was using the Adonis cookie auth system, so I can keep it.

Comment (targos) [reply addons/file-sync/routes.ts]: but what makes it public in one case and not the other?

Comment (maasencioh) [reply addons/file-sync/routes.ts]: get is using a hashed cookie from adonis, post is using a raw token, therefore it needs to be hidden in the body. If you want I can delete the get one

Comment (targos) [reply addons/file-sync/routes.ts]: Please keep get and use an authorization header with the token.

Comment (maasencioh) [reply addons/file-sync/routes.ts]: even if it's not hashed?

Comment (targos) [reply addons/file-sync/routes.ts]: I don't know what you are afraid of, but there is nothing more secure about the body of a request, compared to its headers.

File: app/Validators/MeasurementValidator.ts [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
   public constructor(protected ctx: HttpContextContract) {}
 
   public schema = schema.create({
-    file: schema.file(),
-    filename: schema.string(),
+    collection: schema.string(),
     eventId: schema.string(),
     sampleId: schema.string(),
+    file: schema.file.optional(),
+    filename: schema.string.optional(),
```
Comment (targos) [inline app/Validators/MeasurementValidator.ts]: I'm not sure we need this field? We could use the file name passed with the `file` field.

**Review by targos — APPROVED** [review]

File: app/Validators/MeasurementValidator.ts (hunk 3 above)
Comment (maasencioh) [reply app/Validators/MeasurementValidator.ts]: Sorry, I forgot that we had already discussed that, good catch

**Pull request merged**